src/schema/__tests__/partner_show_events.test.ts

```diff
@@ -0,0 +1,154 @@
+import { runQuery } from "test/utils"
+import gql from "lib/gql"
+import moment from "moment"
+import { ResolverContext } from "types/graphql"
+
+// This time-bomb spec will fail starting at some defined point in the future,
+// indicating that we should consider reverting this behavior of sending
+// different raw date representations to different versions of Eigen
+// (in the hope that usage of older versions has dropped below some minimum threshold).
+
+it("is not yet time to rethink this UA-sniffing behavior for resolving dates", () => {
+  const today = moment()
+  const deadline = moment("2020-10-01") // about 18 months after adding this behavior
+
+  let itIsTimeToRethinkThis = today.isAfter(deadline)
+
+  expect(itIsTimeToRethinkThis).toBe(false)
+})
+
+// In the meantime, here is the custom date resolving behavior
+
+describe("date resolving", () => {
+  const showData = {
+    id: "helwaser-gallery-anton-ginzburg-views",
+    displayable: true,
+    partner: {
+      id: "helwaser-gallery",
+    },
+    events: [
+      {
+        event_type: "Opening Reception",
+        start_at: "2019-03-28T18:00:00+00:00",
+        end_at: "2019-03-28T21:00:00+00:00",
+      },
+    ],
+  }
+
+  const mockLoader = jest.fn(() => Promise.resolve(showData))
+
+  const query = gql`
+    {
+      show(id: "helwaser-gallery-anton-ginzburg-views") {
+        events {
+          start_at
+          end_at
+        }
+      }
+    }
+  `
+
+  let context: Partial<ResolverContext>
+
+  beforeEach(() => {
+    context = {
+      showLoader: mockLoader,
+      partnerShowLoader: mockLoader,
+    }
+  })
+
+  describe("default behavior", () => {
+    it("returns dates with UTC offset", async () => {
+      context.userAgent = "some browser"
+
+      const data = await runQuery(query, context)
+      expect(data).toEqual({
+        show: {
+          events: [
+            {
+              end_at: "2019-03-28T21:00:00+00:00",
+              start_at: "2019-03-28T18:00:00+00:00",
+            },
+          ],
+        },
+      })
+    })
+  })
+
+  describe("with Eigen 4.x", () => {
+    it("truncates the UTC offset", async () => {
```

Comment:
Why are we doing that, instead of only those with the new show view?

Reply:
@kierangillen can confirm but our understanding was that the new Show and Fair pages were released to users before the major version bump that actually introduced the City Guide functionality.

@orta is this something you can confirm? (Other folks who would know are OOO at the moment)

Reply:
Right, I do recall these rolling out before v5, but not in eg v4.0.0

Reply:
Good point! I'm happy to tighten up the version spec here, if we can figure out which versions are affected.

Reply:
Looks like it was 4.4.0 https://github.com/artsy/eigen/blob/27482286c7b29222228192fad732769bd7e61cf7/CHANGELOG.yml#L20